creator-node/test/contentBlacklist.test.js

```diff
@@ -0,0 +1,375 @@
+      .post('/blacklist/delete')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    const user = await models.ContentBlacklist.findOne({ where: { id, type } })
+    assert.deepStrictEqual(user, null)
+    assert.deepStrictEqual(await BlacklistManager.userIdIsInBlacklist(id), 0)
+  })
+
+  it('should return success when removing a track that does not exist', async () => {
+    const id = generateRandomNaturalNumber()
+    const type = BlacklistManager.getTypes().track
+    const { signature, timestamp } = generateTimestampAndSignature({ type, id }, DELEGATE_PRIVATE_KEY)
+
+    await request(app)
+      .post('/blacklist/delete')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    const track = await models.ContentBlacklist.findOne({ where: { id, type } })
+    assert.deepStrictEqual(track, null)
+    assert.deepStrictEqual(await BlacklistManager.trackIdIsInBlacklist(id), 0)
+  })
+
+  it('should ignore duplicate add for track', async () => {
+    const id = generateRandomNaturalNumber()
+    const type = BlacklistManager.getTypes().track
+    const { signature, timestamp } = generateTimestampAndSignature({ type, id }, DELEGATE_PRIVATE_KEY)
+
+    await request(app)
+      .post('/blacklist/add')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    await request(app)
+      .post('/blacklist/add')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    const tracks = await models.ContentBlacklist.findAll({ where: { id, type } })
+    assert.deepStrictEqual(tracks.length, 1)
+    const track = tracks[0]
+    assert.deepStrictEqual(track.id, id)
+    assert.deepStrictEqual(track.type, type)
+    assert.deepStrictEqual(await BlacklistManager.trackIdIsInBlacklist(track.id), 1)
+  })
+
+  it('should ignore duplicate add for user', async () => {
+    const id = generateRandomNaturalNumber()
+    const type = BlacklistManager.getTypes().user
+    const { signature, timestamp } = generateTimestampAndSignature({ type, id }, DELEGATE_PRIVATE_KEY)
+
+    await request(app)
+      .post('/blacklist/add')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    await request(app)
+      .post('/blacklist/add')
+      .query({ type, id, timestamp, signature })
+      .expect(200)
+
+    const users = await models.ContentBlacklist.findAll({ where: { id, type } })
+    assert.deepStrictEqual(users.length, 1)
+    const user = users[0]
+    assert.deepStrictEqual(user.id, id)
+    assert.deepStrictEqual(user.type, type)
+    assert.deepStrictEqual(await BlacklistManager.userIdIsInBlacklist(user.id), 1)
+  })
+
+  // Tests that should return error responses
+
+  it('should throw error if query params does not contain all necessary keys', async () => {
+    const id = generateRandomNaturalNumber()
+    const type = BlacklistManager.getTypes().track
+
+    await request(app)
+      .post('/blacklist/add')
+      .query({ type, id })
```

Comment:
can you add a test case for an incorrect signature, but make sure you pass something in? that's the scariest thing to test